java-checks/src/main/java/org/sonar/java/checks/xml/web/ValidationFiltersCheck.java

```diff
@@ -39,14 +39,15 @@ public void scanWebXml(XmlFile file) {
     evaluateAsList(filterNamesFromFilterExpression, file.getNamespaceUnawareDocument())
       .forEach(node -> {
         String filterName = getStringValue(node);
-        if (!filtersInMapping.contains(filterName)) {
+        if (!filterName.isEmpty() && !filtersInMapping.contains(filterName)) {
           reportIssue(node, "\"" + filterName + "\" should have a mapping.");
         }
       });
   }
 
   private static String getStringValue(Node node) {
-    return node.getFirstChild().getNodeValue();
+    Node firstChild = node.getFirstChild();
+    return (firstChild == null) ? "" : firstChild.getNodeValue();
```

Comment:
you didn't add test for this case

And also it's better to use Optional instead of empty string